### PR TITLE
Limit Sofort description field to 40 characters

### LINKED ISSUE #295: Sofort payment fails for long descriptions
In Sofort, the `description` is not allowed to exceed 40 characters. Otherwise, the API returns `200 OK` (*sigh*…), together with an error in the body.

### PR DESCRIPTION
**Pull request by Natureshadow**
Fixes #295

### CODE REVIEW

File: payments/sofort/__init__.py [hunk 1]
```diff
@@ -56,7 +56,9 @@ def get_form(self, payment, data=None):
                 "interface_version": "django-payments",
                 "amount": payment.total,
                 "currency": payment.currency,
-                "description": payment.description,
+                "description": payment.description
+                if len(payment.description) <= 40
+                else (payment.description[:37] + "..."),
```
Comment (WhyNotHugo) [inline payments/sofort/__init__.py:61]: **minor nitpick**: Is the `…` character accepted by Sofort? Note that it's only one character, so you can leave two more digits.

**Review by WhyNotHugo — COMMENTED** [review]
LGTM

**Comment by Natureshadow** [comment]
> *minor nitpick*: Is the *…* character accepted by Sofort? Note that it's only one character, so you can leave two more digits.

Being a fan of the ellipsis character as well, I did consider this, but decided against it.

Given that Sofort do not seem to be the best at building reasonable APIs, let alone handle errors reasonably, I did not want to take the chance that even if it works now, someone doesn't blindly mess up in the future (deciding to count bytes instead of characters, or something).

**Comment by WhyNotHugo** [comment]
That's reasonable.

**Comment by WhyNotHugo** [comment]
Merged, thanks.

**Pull request closed without merging**